backend/src/ee/services/pam-resource/aws-iam/aws-iam-federation.ts

```diff
@@ -0,0 +1,209 @@
+    const pamStsClient = new STSClient({
+      region: AWS_STS_DEFAULT_REGION,
+      useFipsEndpoint: crypto.isFipsModeEnabled(),
+      sha256: CustomAWSHasher,
+      credentials: {
+        accessKeyId: pamRoleCredentials.Credentials.AccessKeyId!,
+        secretAccessKey: pamRoleCredentials.Credentials.SecretAccessKey!,
+        sessionToken: pamRoleCredentials.Credentials.SessionToken
+      }
+    });
+
+    await pamStsClient.send(
+      new AssumeRoleCommand({
+        RoleArn: targetRoleArn,
+        RoleSessionName: `infisical-pam-target-validation-${Date.now()}`,
+        DurationSeconds: AWS_STS_MIN_DURATION_SECONDS,
+        ExternalId: projectId
+      })
+    );
+
+    return true;
+  } catch {
+    return false;
+  }
+};
+
+/**
+ * Assumes the target role and generates a federated console sign-in URL.
+ */
+export const generateConsoleFederationUrl = async ({
+  connectionDetails,
+  targetRoleArn,
+  roleSessionName,
+  projectId,
+  sessionDuration
+}: {
+  connectionDetails: TAwsIamResourceConnectionDetails;
+  targetRoleArn: string;
+  roleSessionName: string;
+  projectId: string;
+  sessionDuration: number;
+}): Promise<{ consoleUrl: string; expiresAt: Date }> => {
+  const stsClient = createStsClient();
+
+  // First assume the PAM role
+  const pamRoleCredentials = await stsClient.send(
+    new AssumeRoleCommand({
+      RoleArn: connectionDetails.roleArn,
+      RoleSessionName: `infisical-pam-${Date.now()}`,
+      DurationSeconds: sessionDuration,
+      ExternalId: projectId
+    })
+  );
+
+  if (!pamRoleCredentials.Credentials) {
+    throw new Error("Failed to assume PAM role");
+  }
+
+  // Role chaining: use PAM role credentials to assume the target role
+  const pamStsClient = new STSClient({
+    region: AWS_STS_DEFAULT_REGION,
+    useFipsEndpoint: crypto.isFipsModeEnabled(),
+    sha256: CustomAWSHasher,
+    credentials: {
+      accessKeyId: pamRoleCredentials.Credentials.AccessKeyId!,
+      secretAccessKey: pamRoleCredentials.Credentials.SecretAccessKey!,
+      sessionToken: pamRoleCredentials.Credentials.SessionToken
+    }
+  });
+
+  const targetRoleCredentials = await pamStsClient.send(
+    new AssumeRoleCommand({
+      RoleArn: targetRoleArn,
+      RoleSessionName: roleSessionName,
+      DurationSeconds: sessionDuration,
+      ExternalId: projectId
+    })
+  );
+
+  if (!targetRoleCredentials.Credentials) {
+    throw new Error("Failed to assume target role");
+  }
+
+  const { AccessKeyId, SecretAccessKey, SessionToken, Expiration } = targetRoleCredentials.Credentials;
+
+  // Generate federation URL
+  const sessionJson = JSON.stringify({
+    sessionId: AccessKeyId,
+    sessionKey: SecretAccessKey,
+    sessionToken: SessionToken
+  });
+
+  const federationEndpoint = "https://signin.aws.amazon.com/federation";
+
+  const signinTokenUrl = `${federationEndpoint}?Action=getSigninToken&Session=${encodeURIComponent(sessionJson)}`;
+
+  const tokenResponse = await fetch(signinTokenUrl);
```

Comment:
should we use request here instead? 
<img width="874" height="493" alt="image" src="https://github.com/user-attachments/assets/26b24fc8-63f7-45f3-bf9a-d2879902ae91" />